pubsub/gochannel/pubsub.go

```diff
@@ -80,9 +80,11 @@ func NewGoChannel(config Config, logger watermill.LoggerAdapter) *GoChannel {
 //
 // Messages may be persisted or not, depending of persistent attribute.
 func (g *GoChannel) Publish(topic string, messages ...*message.Message) error {
+	g.closedLock.Lock()
```

Comment:
maybe also getter for closed?

Reply:
09777ebf2ea3533a1bda3900aa288b5ab983ada1